### PR TITLE
Update mediawiki to 1.39.7

### PR DESCRIPTION
**Pull request by deer-wmde**
https://lists.wikimedia.org/hyperkitty/list/mediawiki-announce@lists.wikimedia.org/thread/MHGBY3I2OCL7IMUL6F6NUHEWXAZ3DWFE/
https://phabricator.wikimedia.org/T361202

- **update wikiman.yaml to mw 1.39.7**
- **run wikiman**
- **run sync.sh**

### CODE REVIEW

**Comment by deer-wmde** [comment]
*seems* to be related to some change in the Math extension: https://github.com/wbstack/mediawiki/pull/439/files#diff-9ed2b2e3b9657cb4a1d52bfb247b0c11d6b0cc8523911de3632c9f791a054bb2

```
{
  "id": "a4c957e6698c95e98f42432a",
  "type": "ErrorException",
  "file": "/var/www/html/w/includes/Rest/Router.php",
  "line": 186,
  "message": "PHP Warning: filemtime(): stat failed for /var/www/html/w//var/www/html/w/extensions/Math/popupRestRoutes.json",
  "code": 0,
  "url": "/w/rest.php",
  "caught_by": "mwe_handler",
  "backtrace": [
    {
      "function": "handleError",
      "class": "MWExceptionHandler",
      "type": "::",
      "args": [
        "integer",
        "string",
        "string",
        "integer",
        "array"
      ]
    },
    {
      "file": "/var/www/html/w/includes/Rest/Router.php",
      "line": 186,
      "function": "filemtime",
      "args": [
        "string"
      ]
    },
    {
      "file": "/var/www/html/w/includes/Rest/Router.php",
      "line": 221,
      "function": "getRoutesFromFiles",
      "class": "MediaWiki\\Rest\\Router",
      "type": "->",
      "args": []
    },
    {
      "file": "/var/www/html/w/includes/Rest/Router.php",
      "line": 240,
      "function": "getAllRoutes",
      "class": "MediaWiki\\Rest\\Router",
      "type": "->",
      "args": []
    },
    {
      "file": "/var/www/html/w/includes/Rest/Router.php",
      "line": 363,
      "function": "getMatchers",
      "class": "MediaWiki\\Rest\\Router",
      "type": "->",
      "args": []
    },
    {
      "file": "/var/www/html/w/includes/Rest/EntryPoint.php",
      "line": 170,
      "function": "execute",
      "class": "MediaWiki\\Rest\\Router",
      "type": "->",
      "args": [
        "MediaWiki\\Rest\\RequestFromGlobals"
      ]
    },
    {
      "file": "/var/www/html/w/includes/Rest/EntryPoint.php",
      "line": 135,
      "function": "execute",
      "class": "MediaWiki\\Rest\\EntryPoint",
      "type": "->",
      "args": []
    },
    {
      "file": "/var/www/html/w/rest.php",
      "line": 35,
      "function": "main",
      "class": "MediaWiki\\Rest\\EntryPoint",
      "type": "::",
      "args": []
    }
  ],
  "@type": "type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent",
  "severity": "error",
  "serviceContext": {
    "service": "WBaaS MediaWiki",
    "version": "1.0.0"
  }
}

```

File: dist/extensions/Math/extension.json [hunk 1]
```diff
@@ -351,17 +354,5 @@
 	"ServiceWiringFiles": [
 		"ServiceWiring.php"
 	],
-	"RestRoutes": [
```
Comment (deer-wmde) [inline dist/extensions/Math/extension.json:354]: is this causing the error we see in CI?

Comment (deer-wmde) [reply dist/extensions/Math/extension.json:354]: related: https://phabricator.wikimedia.org/T362027

File: wikiman.yaml [hunk 2]
```diff
@@ -72,6 +72,7 @@ extensions:
     repoName: wikimedia/mediawiki-extensions-Kartographer
   - name: Math
     repoName: wikimedia/mediawiki-extensions-Math
+    repoRef: 2822d35380c87960c56111be025fd2c5817e919e
```
Comment (tarrow) [inline wikiman.yaml:75]: I read your comment and the ticket but TBH I failed to understand why we need to pin back to this version; would we be better to either add a patch or actually backport the change to the repo upstream?

Comment (deer-wmde) [reply wikiman.yaml:75]: Solely to prioritize the deployment of the security fix, that cleanup can happen in a follow up PR IMO

File: wikiman.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
 mediawikiVersion: "1.39"
-mediawikiRepoRef: 3406e1b7d014d22f2a68237ebf447a88b5ac74ed
+mediawikiRepoRef: ec2d725f9876b663f5ed736340f54814c023431c
```
Comment (tarrow) [inline wikiman.yaml:3]: I wonder if we would rather remove this? I think we'd be fine always getting the latest 1.39 version.

File: wikiman.yaml (hunk 2 above)
Comment (deer-wmde) [reply wikiman.yaml]: using up-do-date code resolved this

**Comment by deer-wmde** [comment]
> Looks fantastic to me; I'm happy that generally "unpinning" and using master/main of everything where possible seems to be a pattern that's working. Thanks so much for the continued effort to get this over the line!

Yeah I'm also glad that resolved it, thanks for pointing me in the right direction!

**Pull request merged**